src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -0,0 +1,168 @@
+            if (log == null) throw new ArgumentNullException("log");
+            if (svcProvider == null) throw new ArgumentNullException("svcProvider");
+
+            logger = log;
+            serviceProvider = svcProvider;
+            adapterConfig = new EventHubStreamProviderConfig(providerName);
+            adapterConfig.PopulateFromProviderConfig(providerConfig);
+
+            // if no event hub settings type is provided, use EventHubSettings and get populate settings from providerConfig
+            if (adapterConfig.EventHubSettingsType == null)
+            {
+                adapterConfig.EventHubSettingsType = typeof (EventHubSettings);
+            }
+
+            hubSettings = serviceProvider.GetService(adapterConfig.EventHubSettingsType) as IEventHubSettings;
+            if (hubSettings == null)
+            {
+                throw new ArgumentOutOfRangeException("providerConfig", "EventHubSettingsType not valid.");
+            }
+
+            // if settings is an EventHubSettings class, populate settings from providerConfig
+            var settings = hubSettings as EventHubSettings;
+            if (settings != null)
+            {
+                settings.PopulateFromProviderConfig(providerConfig);
+            }
+
+            receivers = new ConcurrentDictionary<QueueId, EventHubAdapterReceiver>();
+            client = EventHubClient.CreateFromConnectionString(hubSettings.ConnectionString, hubSettings.Path);
+        }
+
+        public async Task<IQueueAdapter> CreateAdapter()
+        {
+            partitionIds = await GetPartitionIdsAsync();
+            streamQueueMapper = new EventHubQueueMapper(partitionIds, adapterConfig.StreamProviderName);
+            return this;
+        }
+
+        public IQueueAdapterCache GetQueueAdapterCache()
+        {
+            return this;
+        }
+
+        public IStreamQueueMapper GetStreamQueueMapper()
+        {
+            //TODO: CreateAdapter must be called first.  Figure out how to safely enforce this
+            return streamQueueMapper;
+        }
+
+        public Task<IStreamFailureHandler> GetDeliveryFailureHandler(QueueId queueId)
+        {
+            //TODO: Add a queue specific default failure handler with reasonable error reporting.
+            //TODO: Try to get failure handler from service provider so users can inject their own.
+            return Task.FromResult(streamFailureHandler ?? (streamFailureHandler = new NoOpStreamDeliveryFailureHandler()));
+        }
+
+        public Task QueueMessageBatchAsync<T>(Guid streamGuid, string streamNamespace, IEnumerable<T> events, StreamSequenceToken token,
+            Dictionary<string, object> requestContext)
+        {
+            if (token != null)
+            {
+                throw new ArgumentException("Event Hub stream provider currently does not support non-null StreamSequenceToken.", "token");
```

Comment:
Agreed.  Conceptually, if a user passes an application level sequence token, we should respect it, we simply haven't worked out those details yet, so not implemented would be better.